src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCSVFileRecord.java

```diff
@@ -421,6 +421,40 @@ public Object[] getRowData() throws SQLServerException {
                             break;
                         }
 
+                        case microsoft.sql.Types.VECTOR: {
+                            int dimensionCount = cm.precision;
+                        
+                            if (dimensionCount <= 0) {
+                                throw new SQLServerException("Invalid vector dimension count.", null, 0, null);
+                            }
+                        
+                            String vectorData = data[pair.getKey() - 1].trim();
+                            microsoft.sql.Vector.VectorDimensionType vectorDimensionType = microsoft.sql.Vector.getVectorDimensionType(cm.scale);
+                        
+                            if ("NULL".equalsIgnoreCase(vectorData) || "\"NULL\"".equalsIgnoreCase(vectorData)) {
+                                dataRow[pair.getKey() - 1] = new microsoft.sql.Vector(dimensionCount, vectorDimensionType, null);
+                            } else {
+                                // Remove enclosing brackets or braces if present
+                                if ((vectorData.startsWith("[") && vectorData.endsWith("]")) || 
+                                    (vectorData.startsWith("{") && vectorData.endsWith("}"))) {
```

Comment:
The logic for removing brackets/braces is duplicated. Consider extracting it into a helper method for reusability. Ex:
private String removeEnclosingBrackets(String data) {
    if ((data.startsWith("[") && data.endsWith("]")) || (data.startsWith("{") && data.endsWith("}"))) {
        return data.substring(1, data.length() - 1);
    }
    return data;
}